src/snowflake/connector/errors.py

```diff
@@ -152,6 +152,7 @@ def send_exception_telemetry(
         ):
             # Send with in-band telemetry
             telemetry_data[TelemetryField.KEY_TYPE] = TelemetryField.SQL_EXCEPTION
+            telemetry_data[TelemetryField.KEY_SOURCE] = CLIENT_NAME
```

Comment:
You should be able to overwrite this variable with an argument and when that's not given the do this as the default value